src/client/Client.cc

```diff
@@ -1602,6 +1602,23 @@ int Client::make_request(MetaRequest *request,
   return r;
 }
 
+void Client::unregister_request(MetaRequest *req)
+{
+  mds_requests.erase(req->tid);
+  if (req->tid == oldest_tid) {
+    oldest_tid = 0;
```

Comment:
Why reset this to zero? It's not needed for the following assignment checks, and if there are no entries in mds_requests it won't ever get updated to a non-zero value. We don't want that!